typing/subst.mli

```diff
@@ -54,11 +54,27 @@ val extension_constructor:
         t -> extension_constructor -> extension_constructor
 val class_declaration: t -> class_declaration -> class_declaration
 val cltype_declaration: t -> class_type_declaration -> class_type_declaration
-val modtype: t -> module_type -> module_type
-val signature: t -> signature -> signature
-val signature_item: t -> signature_item -> signature_item
-val modtype_declaration: t -> modtype_declaration -> modtype_declaration
-val module_declaration: t -> module_declaration -> module_declaration
+
+(*
+   When applied to a signature item, a substitution not only modifies the types
+   present in its declaration, but also refreshes the identifier of the item.
+   Effectively this creates new declarations, and so one should decide what the
+   scope of this new declaration should be.
+
+   This is decided by the [scoping] argument passed to the following functions.
+*)
+
+type scoping =
+  | Keep
+  | Make_local
+  | Rescope of int
+
```

Comment:
Could you provide a rationale for which one should be used where?
Something like: inside the environment, and when extracting from it, one should use the scope of the path of the enclosing module, etc ...
Currently this looks a bit like magic.
Even if you don't yet understand fully why this has to be this way, this would give a hint.